common/src/configuration.rs

```diff
@@ -67,6 +67,9 @@ pub fn load_configuration(bootstrap: &ConfigBootstrap) -> Result<Config, String>
 
 /// Installs a new configuration file template, copied from `rincewind-simple.toml` to the given path.
 pub fn install_default_config_file(path: &Path) -> Result<(), std::io::Error> {
+    #[cfg(target_os = "windows")]
+    let source = include_str!("../../config/presets/rincewind-simple.toml");
```

Comment:
We can't use paths outside of the crate -- the package won't compile.

Best other approach is to move the config folder into common (and update the config_generator code as well, which depends on this)

Reply:
Hi Cayle, I do not think this is the issue as far as I can tell. The relative path change is only required for the compiler in Windows vs. Linux. The config generator code does not need to change at all, and the config files can stay where they are. The function `install_default_config_file` is only called in `common/src/lib.rs`
```
        if install {
            println!("Installing new config file at {}", config.to_str().unwrap_or("[??]"));
            install_configuration(&config, configuration::install_default_config_file);
        }
```
and tested in `test_bootstrap_config_from_cli_and_load_configuration`.

The compiler gives the following error if `"../presets/rincewind-simple.toml"` is used in Windows:

```
error: couldn't read common\src\../presets/rincewind-simple.toml: The system cannot find the path specified.
```

However, using `"../../config/presets/rincewind-simple.toml"` works in both Windows and Ubuntu Linux, just not sure about OSX. Running with `cargo` in the terminal:
```
cargo run --bin tari_base_node --release -- --init --create_id
```
or running from the command line:
```
tari_base_node --init --create_id
```

gives Linux console output:
```
$ Tari Base Node
$ Copyright 2019-2020. The Tari Community
$ Version 0.0.10.build-8560aab9

Press Ctrl-C to quit..
Creating "/home/pluto/.tari/"
Installing new config file at /home/pluto/.tari/config.toml
Installing new logfile configuration at /home/pluto/.tari/log4rs.yml
Initializing logging according to "/home/pluto/.tari/log4rs.yml"
Running migration 20190626130555
Running migration 20191030084148
Running migration 20191120090620
Running migration 20191126105357
Running migration 20191126120903
```

and Windows console output:
```
$ Tari Base Node
$ Copyright 2019-2020. The Tari Community
$ Version 0.0.10.build-8560aab9

Press Ctrl-C to quit..
Installing new config file at C:\Users\pluto\.tari\config.toml
Installing new logfile configuration at C:\Users\pluto\.tari\log4rs.yml
Initializing logging according to "C:\\Users\\pluto\\.tari\\log4rs.yml"
Running migration 20190626130555
Running migration 20191030084148
Running migration 20191120090620
Running migration 20191126105357
Running migration 20191126120903
```

In both cases `rincewind-simple.toml` is copied to `/home/pluto/.tari/config.toml` or `C:\Users\pluto\.tari\config.toml`.  

We could probably just change `"../presets/rincewind-simple.toml"` to `"../../config/presets/rincewind-simple.toml"`, assuming it will also work in OSX.

Reply:
Try running `cargo publish` on your change...